library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/DispatcherLocationProvider.java

```diff
@@ -190,12 +191,34 @@ void continueWithDefaultProviders() {
                 getListener().onLocationFailed(FailType.GOOGLE_PLAY_SERVICES_NOT_AVAILABLE);
             }
         } else {
-            LogUtils.logI("Attempting to get location from default providers...");
-            setLocationProvider(getSourceProvider().createDefaultLocationProvider());
+            if(getConfiguration().defaultProviderConfiguration().useAlternativeDefaultProvider()){
+                LogUtils.logI("Attempting to get location from default providers using Alternative Default Location Provider...");
+                setLocationProvider(getSourceProvider().createAlternativeDefaultLocationProvider());
+            } else {
+                LogUtils.logI("Attempting to get location from default provider using Default Location Provider...");
+                setLocationProvider(getSourceProvider().createDefaultLocationProvider());
+            }
             activeProvider.get();
         }
     }
 
+    @Override
+    public void onStartFallbackProvider() {
+        int failType;
+        //Cancel Google Play Service actions
+        cancel();
+        if(getConfiguration().googlePlayServicesConfiguration().useFallbackProviderAfterConnectionFailed()
+                && getConfiguration().defaultProviderConfiguration() != null){
+            failType = FailType.GOOGLE_PLAY_SERVICES_CONNECTION_FAIL_START_FALLBACK;
+            continueWithDefaultProviders();
```

Comment:
Here you call `continueWithDefaultProviders` but then you call `onLocationFailed`.
That's not correct, if location tracking will be still on progress, listener should not receive failing.